src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -187,56 +181,23 @@ public List<GitCommit> GetCommitHistory(string file, string committish = null)
             return result.Select(c => c.GitCommit).ToList();
         }
 
-        public Task SaveCache()
+        public void Dispose()
         {
-            if (!_cacheUpdated || string.IsNullOrEmpty(_cacheFilePath))
+            var repo = Interlocked.Exchange(ref _repo, IntPtr.Zero);
+            if (repo != IntPtr.Zero)
             {
-                return Task.CompletedTask;
+                git_repository_free(_repo);
             }
-
-            PathUtility.CreateDirectoryFromFilePath(_cacheFilePath);
-
-            return ProcessUtility.WriteFile(_cacheFilePath, stream =>
-            {
-                using (var writer = new BinaryWriter(stream))
-                {
-                    // Create a snapshot of commit cache to ensure count and items matches.
-                    var commitCache = _commitCache.ToList();
-
-                    writer.Write(commitCache.Count);
-                    foreach (var (file, value) in commitCache)
-                    {
-                        lock (value)
-                        {
-                            writer.Write(file);
-                            writer.Write(Math.Min(value.Count, MaxCommitCacheCountPerFile));
-
-                            var lruValues = value.OrderBy(pair => pair.Value.lruOrder).Take(MaxCommitCacheCountPerFile);
-
-                            foreach (var ((commit, blob), (commitHistory, _)) in lruValues)
-                            {
-                                writer.Write(commit);
-                                writer.Write(blob);
-                                writer.Write(commitHistory.Length);
-
-                                foreach (var sha in commitHistory)
-                                {
-                                    writer.Write(sha);
-                                }
-                            }
-                        }
-                    }
-                }
-            });
         }
 
-        public void Dispose()
+        public Task Update()
```

Comment:
Can we still call this `SaveCache`? We just moved the implementation to `GitCommitCacheProvider.SaveCache`

Reply:
ok